src/libcore/str/mod.rs

```diff
@@ -2142,7 +2141,7 @@ pub trait StrExt {
     #[stable(feature = "core", since = "1.6.0")]
     fn is_empty(&self) -> bool;
     #[stable(feature = "core", since = "1.6.0")]
-    fn parse<'a, T: TryFrom<&'a str>>(&'a self) -> Result<T, T::Error>;
```

Comment:
Why is this reverted? This allows for fewer conversions. With `TryFrom` you can make the error lifetime depend on the string lifetime, but not with `FromStr`.